src/bootstrap-window.js

```diff
@@ -81,22 +81,18 @@
 
 		window.document.documentElement.setAttribute('lang', locale);
 
-		// do not advertise AMD to avoid confusing UMD modules loaded with nodejs (TODO@sandbox non-sandboxed only)
-		if (!sandbox) {
-			window['define'] = undefined;
-		}
-
 		// replace the patched electron fs with the original node fs for all AMD code (TODO@sandbox non-sandboxed only)
 		if (!sandbox) {
-			require.define('fs', ['original-fs'], function (originalFS) { return originalFS; });
+			require.define('fs', [], function () { return require.__$__nodeRequire('original-fs'); });
 		}
 
 		window['MonacoEnvironment'] = {};
 
 		const loaderConfig = {
-			baseUrl: `${uriFromPath(configuration.appRoot)}/out`,
+			baseUrl: 'vscode-file://localhost/' + `${uriFromPath(configuration.appRoot)}/out`.substr(8),
```

Comment:
This logic should go into the `uriFromPath` method which is now part of `bootstrap-window.js` and thus safe to be changed.